lib/src/folding/matchers/default_foldable_block_matcher.dart

```diff
@@ -36,7 +40,7 @@ class FoldableBlockMatcher {
       if (_match(oldBlock: oldBlock, newBlock: newBlock)) {
         oldToNew[oldBlock] = newBlock;
       } else {
-        break;
+        continue;
```

Comment:
I think we need to discuss this.

This fixes the issue with folded block:
```java
class MyClass {
  /// Calculates the factorial of the number.
  /// The number must be >= 0.
  static int factorial(int n) {
    if (n == 0) {
      return 1;
    }

    return n * factorial(n - 1);
  }

  public static void main(String[] args) {
    int num = 5;
    System.out.println("Factorial of " + num + " is " + factorial(5));
  }
}
```
WHEN break instead of continue.
1. fold at 4th line.
2. move cursor to the end of 13th line.
3. type any character.
RESULT: folded block is unfolded, cursor is moved to the end of the 4th line

Should it be included here?

Reply:
It comes with other bugs..
I think we need to entirely rewrite this logic.